### PR TITLE
Using Profile Client against Profile mock server

### PR DESCRIPTION
**Pull request by Oluane**
## Description

In this PR, I created the GET Profile api route to contact Profile mock server, with the Profile Client.

The `/api/profile/get-profile` api route can be used to get both profile user info and profile addresses at the same time. This endpoint isn't used yet to display info on front side. I let the previous `/api/profile/get-profile` which returned hard-coded values, I simply renamed it to `/api/profile/get-profile-fake` for now. 

The new handler uses newly added web errors to handle Profile's errors.
For now, as it is implemented, if an error occurred in one of the two calls (user info or addresses), we send back no data at all. It's all or nothing. I think this is a topic we should discuss in refinement, to decide what we want to do about that. I will create a task so we can talk about it next time.

The Profile client init is now in an externalized function, inside `/src/configs/`. 
I also added a new key to `configVariables` for `PROFILE_MOCK_SERVER_URL` env var.

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-gf7bf9 Use Profile generated client against the mock server (URL via env_var)](https://app.clickup.com/t/gf7bf9)

**Comment by Oluane** [comment]
@Fuzznimp @Meyclem Do you guys think I should also implement the two update user info/addresses api handlers in this PR or merge that and doing the update part later ? 🤔

**Comment by Fuzznimp** [comment]
Now that I think about it, pretty sure those endpoints do not exist on the OpenAPI client yet 🤔 .

**Comment by Oluane** [comment]
Ok, I think profile user info should already exist, but not address yet you're right.

**Comment by Fuzznimp** [comment]
Then, I would say to do the other one in another PR :).

File: src/pages/api/profile/get-profile.ts [hunk 1]
```diff
@@ -32,62 +41,62 @@ const handler: Handler = async (request, response) => {
...
+    span.setDisclosedAttribute("is profile addresses fetched", true);
+
     response.statusCode = HttpStatus.OK;
-    response.json({ userInfo: mockedUserInfo });
+    response.json({
+      profileData: {
+        profileUserInfo,
+        profileAddresses,
+      },
+    });
```
Comment (Fuzznimp) [inline src/pages/api/profile/get-profile.ts:99]: Maybe we should go with
```ts
response.json({
    profileUserInfo,
    profileAddresses,
  });
```

What do you think?

Comment (Meyclem) [reply src/pages/api/profile/get-profile.ts:99]: I agree

**Review by Fuzznimp — APPROVED** [review]

**Review by Meyclem — COMMENTED** [review]
>For now, as it is implemented, if an error occurred in one of the two calls (user info or addresses), we send back no data at all. It's all or nothing. I think this is a topic we should discuss in refinement, to decide what we want to do about that.

I answered in one on the comment but for more clarity I also answer here.

I think we should have different endpoints for profiles and addresses.
- It will ease the error handling
- we may not want to get all the data each time

**Comment by Meyclem** [comment]
> @Fuzznimp @Meyclem Do you guys think I should also implement the two update user info/addresses api handlers in this PR or merge that and doing the update part later ? 🤔

The profile update is here but not the address part.

**Comment by Fuzznimp** [comment]
> we may not want to get all the data each time

But for now there are no reason to only get all address OR all profile data. Having one endpoint remove on fetch.

File: src/configs/config-variables.ts [hunk 2]
```diff
@@ -54,6 +55,7 @@ const configVariables: ConfigVariables = {
     URI: "",
     APIKey: "",
   },
+  profileMockServerUrl: "",
```
[the remark this answers is not in the document]
Comment (Oluane) [reply src/configs/config-variables.ts:58]: Yep, I totally agree with that

File: src/errors/web-errors.ts [hunk 3]
```diff
@@ -64,6 +74,11 @@ const ERRORS_DATA: Record<string, WebErrorData> = {
     httpStatus: HttpStatus.NOT_FOUND,
     message: "Identity not found",
   },
+  PROFILE_USER_NOT_FOUND: {
+    code: "profile_user_not_found",
+    httpStatus: HttpStatus.NOT_FOUND,
+    message: "Profile user not found",
```
[the remark this answers is not in the document]
Comment (Oluane) [reply src/errors/web-errors.ts:80]: For this one, I took Profile's web errors and added it on Account, and it's send back by get user profile (user-info) endpoint. The naming should be refined indeed.

**Pull request merged**